### DISCUSSION TITLE
Displaying lists of people based on Start Date or End Date custom fields

### DISCUSSION THREAD

**Discussion by bkeplinger** [liedekef/events-made-easy#505] [General]
I have found this plugin to be very useful to display groups of people, such as employees or board members (I use very little of the event functionality, or for that matter, any of the other features of this plugin).

I have been grappling with how to display changing staff lists. I settled on creating a new group for a modified staff list (based on new employees or employees leaving) and using a plugin to create a shortcode controlling the display of the groups. This was clunky, at best, but it worked.

It occurred to me that if I had a start/end dates for people, I could control the display of individuals withing the group, so I added custom fields for People: Start Date and End Date.

If I have a Start Date for a person I want that person to display on or after that date, and if I have an End Date for a person I want that person to display on or before that date (of if it would be easier, I'd choose the End Date to be the day after the final day of employment, so that person would display before that date). So, I guess I want People to act like events in regard to the Start and End dates.

That's where I ran into a road block. I looked through the documentation but I couldn't find how to use these custom fields to control the display of a person. I saw where I could do this with Events, but not with a custom field for People.

Would such a thing be possible? I guess I can do this with JavaScript, but I'd like to do it with the the plugin itself, if possible.

Bruce

**Comment by liedekef** [comment]
The eme_if shortcode works in the people list too, so something like this would work according to me. To show someone that currently started: 
```
[eme_if tag="#_FIELDVALUE{myenddate}" is_empty="1"]
[eme_if2 tag="#_CURDATE{Y-m-d}" gt="#_FIELDVALUE{mystartdate}"] .... [/eme_if2]
[/eme_if]
```
However, moving people leaving the company to another group seems way more efficient to me. EME can display groups of people using templates, so using another plugin to create a shortcode seems a bit weird ...

**Reply by bkeplinger** [reply]
Ah! Thank you!

As far as moving to another group, I want this to happen without anyone's direct input, which to me is the whole idea behind a CMS (or at least a major part of it).

Lastly, I didn't know [eme_if] tags could be nested, which I suspected is what I needed to do. I tried to do that with something something else the other day (I can't remember what, but it didn't involve this problem) and it didn't work as I expected. I'll give it another go.

Many thanks,
Bruce

**Reply by bkeplinger** [reply]
Ah! I see! [eme_if] [eme2_if]...

**Comment by liedekef** [comment]
> Ah! I see! [eme_if] [eme2_if]...

indeed, EME supports 15 levels of nesting (there's no such thing as if-then-else in shortcodes ...)
 
> As far as moving to another group, I want this to happen without anyone's direct input, which to me is the whole idea behind a CMS (or at least a major part of it).

A CMS (content management system) is not really meant for what you're saying, but by using plugins WP has become so much more. That's also the reason I've built people/group and custom people fields into the system. You can even create a group and have fields only appear for people in that group (that's what I'm using for volunteer management and also for instructors and management of a non-profit). I'm glad to see someone else using EME for organisational purposes.

**Comment by bkeplinger** [comment]
Just as this came in, I was trying a simple test, because your example wasn’t working for me.

Here’s what I did:
[eme_if tag="#_FIRSTNAME" value="John"] FN
[eme2_if tag="#_LASTNAME" value="Doe"] LN [/eme2_if]
[/eme_if]
#_FIRSTNAME _LASTNAME

This is what displays:
FN
[eme2_if tag=”Doe” value=”Doe”]LN[/eme2_if] John Doe

I’m clearly doing something wrong. (I updated the plugin earlier this week.)

Bruce

**Reply by liedekef** [reply]
it is eme_if2, not eme2_if (I corrected my comment above, typo)

**Comment by bkeplinger** [comment]
Ah!! (I should have tried that, because I thought it would be more logical than eme2_if.)

**Reply by liedekef** [reply]
When in doubt, check the doc: https://www.e-dynamics.be/wordpress/conditional-tags/
(there's a for-loop there too btw)

Also: avoid replying to github mails, it cpmes out ugly in the discussion threads (I'm a bit wondering as to why they allow email replies but don't clean up the reply ...)

**Comment by bkeplinger** [comment]
I finally got this to work early last month. It took me some trial and error to re-reading of your comments that got me there. 

1. PHP doesn't treat dates like a number, so format matters.
2. The date field must be JavaScript.

This brings up another issue, but it's not related to this, so I'll submit another issue.

Thanks.
